internal/bridge/installation.go

```diff
@@ -157,24 +167,30 @@ func (r *InstallationReconciler) reconcile(ctx context.Context, read *corev1.Sec
 	// Secret which triggers another reconcile.
 	if len(installation.ID) == 0 {
 		if len(self.ID) == 0 {
-			return r.register(ctx, write)
+			return 0, r.register(ctx, write)
 		}
 
 		data := map[string][]byte{}
 		data[KeyBridgeToken], _ = json.Marshal(self.Installation) //nolint:errchkjson
 
-		return r.persist(ctx, write.WithData(data))
+		return 0, r.persist(ctx, write.WithData(data))
 	}
 
-	// When the Secret has an Installation, store it in memory.
-	// TODO: Validate it first; perhaps refresh the AuthObject.
-	if len(self.ID) == 0 {
-		self.Lock()
-		self.Installation = installation
-		self.Unlock()
+	// Read the timestamp from the Secret, if any.
+	var touched time.Time
+	if yaml.Unmarshal(read.Data[KeyBridgeLocalTime], &touched) != nil {
+		touched = time.Time{}
+	}
+
+	// Refresh the AuthObject when there is no Installation in memory,
+	// there is no timestamp, or the timestamp is far away. This writes to
+	// the Secret which triggers another reconcile.
+	if len(self.ID) == 0 || time.Since(touched) > r.Refresh || time.Until(touched) > r.Refresh {
+		return 0, r.refresh(ctx, installation, write)
 	}
 
-	return nil
+	// Trigger another reconcile one interval after the stored timestamp.
+	return wait.Jitter(time.Until(touched.Add(r.Refresh)), 0.1), nil
```

Comment:
🤔 I'm not sure about this line. Do we need jitter here? There is no test/mock clock, so the test needs some fuzz/slack/threshold either way.